src/modules/expression/binop/range.rs

```diff
@@ -59,17 +59,60 @@ impl SyntaxModule<ParserMetadata> for Range {
 impl TranslateModule for Range {
     fn translate(&self, meta: &mut TranslateMetadata) -> String {
         let from = self.from.translate(meta);
-        let to = self.to.translate(meta);
-        if self.neq {
-            let to_neq = if let Some(ExprType::Number(_)) = &self.to.value {
-                to.parse::<isize>().unwrap_or_default().sub(1).to_string()
+        let to = if let Some(to) = self.to.get_integer_value() {
```

Comment:
I'm not sure it's that easy, because I'm using this functionality for `Neg::get_array_index()` as well, and we might want to expand it to more expression types in future.

Reply:
Also, these functions behave recursively; `Expr::get_integer_value()` calls `Neg::get_integer_value()`, which calls `Expr::get_integer_value()` again. It would be hard to implement this in a single function.